doc/specs/#888 - Com Api.md

```diff
@@ -0,0 +1,812 @@
+        check_hresult(CoCreateInstance(CLSID_AppInstaller, nullptr, CLSCTX_LOCAL_SERVER, IID_PPV_ARGS(abiAppInstaller.put())));
+        check_pointer(abiAppInstaller.get());
+        AppInstaller appInstaller{ abiAppInstaller.as<AppInstaller>() };
+        return appInstaller;
+    }
+    InstallOptions CreateInstallOptions()
+    {
+        com_ptr<abi::IInstallOptions> abiInstallOptions = nullptr;
+        check_hresult(CoCreateInstance(CLSID_InstallOptions, nullptr, CLSCTX_LOCAL_SERVER, IID_PPV_ARGS(abiInstallOptions.put())));
+        check_pointer(abiInstallOptions.get());
+        InstallOptions installOptions{ abiInstallOptions.as<winrt::Microsoft::Management::Deployment::InstallOptions>() };
+        return installOptions;
+    }
+    FindPackagesOptions CreateFindOptions()
+    {
+        com_ptr<abi::IFindPackagesOptions> abifindOptions = nullptr;
+        check_hresult(CoCreateInstance(CLSID_FindPackagesOptions, nullptr, CLSCTX_LOCAL_SERVER, IID_PPV_ARGS(abifindOptions.put())));
+        check_pointer(abifindOptions.get());
+        FindPackagesOptions findPackagesOptions{ abifindOptions.as<FindPackagesOptions>() };
+        return findPackagesOptions;
+    } 
+    GetCompositeAppCatalogOptions CreateGetCompositeAppCatalogOptions()
+    {
+        com_ptr<abi::IGetCompositeAppCatalogOptions> abiGetCompositeAppCatalogOptions = nullptr;
+        check_hresult(CoCreateInstance(CLSID_GetCompositeAppCatalogOptions), nullptr, CLSCTX_LOCAL_SERVER, IID_PPV_ARGS(abiGetCompositeAppCatalogOptions.put())));
+        check_pointer(abiGetCompositeAppCatalogOptions.get());
+        GetCompositeAppCatalogOptions getCompositeAppCatalogOptions{ abiGetCompositeAppCatalogOptions.as<GetCompositeAppCatalogOptions>() };
+        return getCompositeAppCatalogOptions;
+    }
+```
+
+## 3.2. Search
+
+The api can be used to search for packages in a catalog known to Windows Package Manager. This can be used to get availability information
+or start an install.
+
+```c++ (C++ish pseudocode)
+
+
+    IAsyncOperation<Package> FindPackageInCatalog(AppCatalog catalog, std::wstring packageId)
+    {
+        FindPackagesOptions findPackagesOptions = CreateFindOptions();
+        PackageMatchFilter filter;
+        filter.IsAdditive = true;
+        filter.Field = PackageMatchField::Id;
+        filter.Type = MatchType::Exact;
+        filter.Value = packageId;
+        findPackagesOptions.Filters().Append(filter);
+        FindPackagesResult findPackagesResult{ co_await catalog.FindPackagesAsync(findPackagesOptions) };
+
+        winrt::IVectorView<ResultMatch> matches = findPackagesResult.Matches();
+        co_return matches.GetAt(0).Package();
+    }
+
+    IAsyncOperation<Package> MainPage::FindPackage()
+    {
+        // Capture the ui thread context.
+        winrt::apartment_context uiThread;
+        co_await winrt::resume_background();
+
+        AppInstaller appInstaller = CreateAppInstaller();

```

Comment:
Since you explicitly jumped to the background thread, you can use the `.get()` synchronous-blocking form of the call here rather than `co_await` here.